### PR TITLE
Fixed banner menu can not be opened

### PR DESCRIPTION
**Pull request by montlikadani**
With this PR the banner menu can be opened, the only issue is that there is no back button and there is no way to retrieve the banner from menu (idk how it works)

### CODE REVIEW

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/Banner.java [hunk 1]
```diff
@@ -64,14 +66,15 @@ public BannerPattern getLastPattern() {
   public ItemStack buildBanner() {
     ItemStack item = XMaterial.WHITE_BANNER.parseItem();
     BannerMeta meta = (BannerMeta) item.getItemMeta();
-    if(item instanceof org.bukkit.block.Banner) {
-      ((org.bukkit.block.Banner) item).setBaseColor(this.color);
-    } else {
-      meta.setBaseColor(this.color);
+
+    if(ServerVersion.Version.isCurrentEqualOrLower(ServerVersion.Version.v1_12_R1)) {
+      meta.setBaseColor(color);
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/Banner.java:71]: difference of item and meta location of setBaseColor now removed?

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java [hunk 2]
```diff
@@ -62,74 +63,81 @@ public BannerMenu(Player player, Banner banner) {
...
       if(ServerVersion.Version.isCurrentEqualOrLower(ServerVersion.Version.v1_12_R1)) {
         item = XMaterial.WHITE_BANNER.parseItem();
         BannerMeta meta = (BannerMeta) item.getItemMeta();
-        if(ServerVersion.Version.isCurrentEqualOrLower(ServerVersion.Version.v1_12_R1)) {
-          meta.setBaseColor(color);
-        } else {
-          ((org.bukkit.block.Banner) item).setBaseColor(color);
-        }
+
+        meta.setBaseColor(color);
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java:77]: higher 1.12 works?

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java [hunk 3]
```diff
@@ -62,74 +63,81 @@ public BannerMenu(Player player, Banner banner) {
...
       ItemStack item = banner.buildBanner();
       BannerMeta meta = (BannerMeta) item.getItemMeta();
-      Pattern pattern = new Pattern(color, banner.getLastPattern().getPatternType());
-      meta.addPattern(pattern);
+
+      meta.addPattern(new Pattern(color, banner.getLastPattern().getPatternType()));
       item.setItemMeta(meta);
+
       gui.addItem(new SimpleClickableItem(item, e -> {
         e.setCancelled(true);
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java:131]: we are already here so e -> event

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java [hunk 4]
```diff
@@ -62,74 +63,81 @@ public BannerMenu(Player player, Banner banner) {
...
+
     for(PatternType pattern : PatternType.values()) {
       ItemStack item = banner.buildBanner();
       BannerMeta meta = (BannerMeta) item.getItemMeta();
       DyeColor color = banner.getColor() == DyeColor.BLACK ? DyeColor.WHITE : DyeColor.BLACK;
+
       meta.addPattern(new Pattern(color, pattern));
       item.setItemMeta(meta);
+
       gui.addItem(new SimpleClickableItem(item, e -> {
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java:107]: we are already here so e -> event

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java [hunk 5]
```diff
@@ -62,74 +63,81 @@ public BannerMenu(Player player, Banner banner) {
...
-        }
+
+        meta.setBaseColor(color);
         item.setItemMeta(meta);
       } else {
-        item = XMaterial.matchXMaterial(color.toString().toUpperCase(Locale.ENGLISH) + "_BANNER").get().parseItem();
+        item = XMaterial.matchXMaterial(color.toString() + "_BANNER").orElse(XMaterial.WHITE_BANNER).parseItem();
       }
+
       gui.addItem(new SimpleClickableItem(item, e -> {
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java:83]: we are already here so e -> event

**Review by Tigerpanzer02 — CHANGES_REQUESTED** [review]

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java (hunk 2 above)
Comment (montlikadani) [reply src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/BannerMenu.java:77]: There is a condition above this `item = XMaterial.WHITE_BANNER.parseItem();`

File: src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/Banner.java (hunk 1 above)
Comment (montlikadani) [reply src/main/java/plugily/projects/buildbattle/handlers/menu/registry/banner/Banner.java:71]: No, the `ItemStack` can not be cast to `org.bukkit.block.Banner`, also `BannerMeta#addPattern` will handles the colours.

**Pull request merged**